src/actions/retriveAlreadyBookedDaysActions.js

```diff
@@ -1,77 +1,89 @@
-const { readFile, parseJSON } = require("../helpers/helpers");
-
-const parseJSONFromFile = path => {
-    return readFile(path)
-        .then(parseJSON)
-        .catch(err => {
-            throw {
-                status: 500,
-                message: "the error was logged and we’ll be checking it shortly"
-            };
-        });
-};
-
-const findBookedDays = (userId, daysOff) => {
-    let daysBooked = [];
-    daysOff.forEach(element => {
-        if (element["userId"] === userId) {
-            daysBooked.push(...element["daysOff"]);
-        }
-    });
-    return daysBooked;
-};
+const DaysOff = require("../models/daysOffModel")
+const User = require("../models/userModel")
+const findBookedDays = (userId) => {
+    return new Promise((resolve, reject) => {
+        findDaysOff = DaysOff.find({ userId: userId }, "daysOff", (err, daysOff) => {
+            if (err) reject(err)
+            resolve(JSON.parse(JSON.stringify(daysOff)))
+        })
+    })
+}
 
-const getAlreadyBookedDays = (userId, daysOff) => {
-    return new Promise((response, reject) => {
-        const bookedDays = findBookedDays(userId, daysOff);
-        if (bookedDays.length === 0)
+const checkUserExistsInDB = (userId) => {
+    return new Promise((resolve, reject) => {
+        let findUser = User.findById(userId, (err, user) => {
+            if (err)
+                reject({
+                    status: 400,
+                    message: "Bad user id"
+                });
+            if (user) resolve();
```

Comment:
Remember the discussion regarding interrupting the flow when you want to

Reply:
Fixed it in latest commits.